### PR TITLE
Install Enterprise via repo

### PR DESCRIPTION
**Pull request by lakrass**
- Add ability to install Vault Enterprise via HashiCorp repo
- Set the default value of the vault_version_repo_suffix variable according to the vault_enterprise variable
- Notify the service restart handler on installation via HashiCorp repo
- Removed check of non-existent variable "vault_install_remote_repo" in tasks/main.yml
- Preparations to release v2.5.9

### CODE REVIEW

File: .gitignore
Comment (sschmittsva) [inline .gitignore:1]: unrelated

File: flake.lock
Comment (sschmittsva) [inline flake.lock:1]: unrelated

File: flake.nix
Comment (sschmittsva) [inline flake.nix:1]: unrelated

File: tasks/install_hashi_repo.yml [hunk 1]
```diff
@@ -65,18 +65,24 @@
         name: "{{ vault_rhsm_repo_id }}"
         state: enabled
 
+- name: Ensure Enterprise package is not present when OSS is desired and vice versa
+  package:
+    name: "{{ 'vault' ~ ('-enterprise' if not (vault_enterprise | bool) else '') }}"
```
Comment (sschmittsva) [inline tasks/install_hashi_repo.yml:70]: "if not" -> "if"?

File: tasks/main.yml [hunk 2]
```diff
@@ -58,7 +58,7 @@
   when:
     - vault_enterprise | bool
     - not vault_install_remotely | bool
-    - not vault_install_remote_repo | bool
+    - not vault_install_hashi_repo | bool
```
Comment (sschmittsva) [inline tasks/main.yml:61]: vault_install_remote_repo does not seem to be used elsewhere. But I'm not sure if we want to remove it in this PR.

File: tasks/main.yml [hunk 3]
```diff
@@ -72,7 +72,6 @@
 - name: Install Vault via HashiCorp repository
   include_tasks: install_hashi_repo.yml
   when:
-    - not vault_enterprise | bool
```
Comment (sschmittsva) [inline tasks/main.yml:75]: Do we really want to remove it here?

**Review by sschmittsva — CHANGES_REQUESTED** [review]
Hi!

Thanks for your PR. I would like to split out the Nix parts to have a minimal PR.

File: tasks/main.yml (hunk 2 above)
Comment (lakrass) [reply tasks/main.yml:61]: I'd remove it in another PR :)

File: tasks/install_hashi_repo.yml (hunk 1 above)
Comment (lakrass) [reply tasks/install_hashi_repo.yml:70]: "not" is correct here 'cause this task is to uninstall enterprise if community/OSS is targeted and vice versa:

vault_enterprise: true -> "vault" should be absent
vault_enterprise: false -> "vault-enterprise" should be absent

File: tasks/main.yml (hunk 3 above)
Comment (lakrass) [reply tasks/main.yml:75]: Yes, because otherwise installation of Vault Enterprise via would not be possible

File: .gitignore
Comment (lakrass) [reply .gitignore:1]: removed

File: flake.lock
Comment (lakrass) [reply flake.lock:1]: removed

File: flake.nix
Comment (lakrass) [reply flake.nix:1]: removed

File: tasks/main.yml (hunk 2 above)
Comment (lakrass) [reply tasks/main.yml:61]: In my opinion we can safely remote this variable here, 'cause it's - as you said - not used anywhere else. Also I stated the removal in the changelog as well as in the PR.

**Comment by lakrass** [comment]
Hey @sschmittsva, 

I included your latest changes from the master in this PR, responded to your comments and fixed a few things I agree with. 
Feel free to check my PR again. 

We should discuss on releasing v2.5.9 with this PR. If we do a release by this PR, we should remove the "- WIP" from the changelog. If you plan other changes before releasing the next version, we could keep the "- WIP".

Thank you for your help! :)

**Comment by lakrass** [comment]
I added another small change concerning the default value of the vault_version_repo_suffix variable

**Comment by lakrass** [comment]
I added two more small changes: 
- set the default value of the vault_version_repo_suffix variable according to the vault_enterprise variable
- notify the service restart handler on installation via HashiCorp repo

File: CHANGELOG.md [hunk 4]
```diff
@@ -3,6 +3,28 @@
 - Add support for proxy configuration
 - Add support for system certificates with `vault_tls_certs_path` and `vault_tls_private_path`
 
+## v2.5.9 - WIP
```
Comment (FalcoSuessgott) [inline CHANGELOG.md]: Remove wip

File: .gitignore [hunk 5]
```diff
@@ -9,4 +9,4 @@ examples/hosts
 files/vault
 files/vault*_SHA256SUMS
 meta/.galaxy_install_info
-examples/roles/ansible-community.ansible-vault
+examples/roles/ansible-community.ansible-vault
```
Comment (FalcoSuessgott) [inline .gitignore]: Trailing line

**Review by FalcoSuessgott — CHANGES_REQUESTED** [review]
2 minor things before merging :)

File: .gitignore (hunk 5 above)
Comment (lakrass) [reply .gitignore]: added

File: CHANGELOG.md (hunk 4 above)
Comment (lakrass) [reply CHANGELOG.md]: removed

**Comment by lakrass** [comment]
Hello @FalcoSuessgott, thank you for your review. :) 
I just fixed the remarks you issued.

**Comment by FalcoSuessgott** [comment]
LGTM Atze :)

**Review by FalcoSuessgott — APPROVED** [review]
